sdk/core/azure-core/azure/core/polling/async_base_polling.py

```diff
@@ -23,24 +23,70 @@
 # IN THE SOFTWARE.
 #
 # --------------------------------------------------------------------------
+from typing import TypeVar, cast, Union
 from ..exceptions import HttpResponseError
 from .base_polling import (
     _failed,
     BadStatus,
     BadResponse,
     OperationFailed,
-    LROBasePolling,
+    _SansIOLROBasePolling,
     _raise_if_bad_http_status_and_method,
 )
+from ._async_poller import AsyncPollingMethod
 from ..pipeline._tools import is_rest
+from .. import AsyncPipelineClient
+from ..pipeline import PipelineResponse
+from ..pipeline.transport import (
+    HttpRequest as LegacyHttpRequest,
+    AsyncHttpTransport,
+    AsyncHttpResponse as LegacyAsyncHttpResponse,
+)
+from ..rest import HttpRequest, AsyncHttpResponse
+
+HttpRequestType = Union[LegacyHttpRequest, HttpRequest]
+AsyncHttpResponseType = Union[LegacyAsyncHttpResponse, AsyncHttpResponse]
+LegacyPipelineResponseType = PipelineResponse[LegacyHttpRequest, LegacyAsyncHttpResponse]
+NewPipelineResponseType = PipelineResponse[HttpRequest, AsyncHttpResponse]
+AsyncPipelineResponseType = PipelineResponse[HttpRequestType, AsyncHttpResponseType]
+HttpRequestTypeVar = TypeVar("HttpRequestTypeVar", bound=HttpRequestType)
+AsyncHttpResponseTypeVar = TypeVar("AsyncHttpResponseTypeVar", bound=AsyncHttpResponseType)
+AsyncPipelineResponseTypeVar = PipelineResponse[HttpRequestTypeVar, AsyncHttpResponseTypeVar]
+
+
+PollingReturnType_co = TypeVar("PollingReturnType_co", covariant=True)
 
 __all__ = ["AsyncLROBasePolling"]
 
 
-class AsyncLROBasePolling(LROBasePolling):
-    """A subclass or LROBasePolling that redefine "run" as async."""
+class AsyncLROBasePolling(
+    _SansIOLROBasePolling[
+        PollingReturnType_co,
+        AsyncPipelineClient[HttpRequestTypeVar, AsyncHttpResponseTypeVar],
+    ],
+    AsyncPollingMethod[PollingReturnType_co],
+):
+    """A base LRO async poller.
+
+    This assumes a basic flow:
+    - I analyze the response to decide the polling approach
+    - I poll
+    - I ask the final resource depending of the polling approach
+
+    If your polling need are more specific, you could implement a PollingMethod directly
+    """
 
-    async def run(self):  # pylint:disable=invalid-overridden-method
+    _initial_response: AsyncPipelineResponseType
+    """Store the initial response."""
+
+    _pipeline_response: AsyncPipelineResponseType
+    """Store the latest received HTTP response, initialized by the first answer."""
```

Comment:
Should these be typed as `AsyncPipelineResponseTypeVar` to be consistent with what we do in sync polling?